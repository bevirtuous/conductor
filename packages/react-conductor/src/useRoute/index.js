import { useContext } from 'react';
import { RouterContext } from '../context';

function useRoute() {
  // This causes the error.
  return useContext(RouterContext);
}

export default useRoute;
