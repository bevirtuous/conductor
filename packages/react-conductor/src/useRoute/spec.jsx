import React from 'react';
import { shallow } from 'enzyme';
import useRoute from './index';

/**
 * 
 */
function MyComponent() {
  const route = useRoute();
  return JSON.stringify(route);
}

describe('useRoute()', () => {
  it('should return the current route', () => {
    const component = shallow((
      <MyComponent />
    ));

    expect(component).toMatchSnapshot();
  });
});
