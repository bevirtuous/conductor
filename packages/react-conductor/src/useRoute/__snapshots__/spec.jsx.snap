// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useRoute() should return the current route 1`] = `"{\\"hi\\":4}"`;
